src/tools/chdman.cpp

```diff
@@ -1462,62 +1466,19 @@ static void compress_common(chd_file_compressor &chd)
 //  to a CUE file
 //-------------------------------------------------
 
-void output_track_metadata(int mode, util::core_file &file, int tracknum, const cdrom_file::track_info &info, const std::string &filename, uint32_t frameoffs, uint64_t discoffs)
+void output_track_metadata(int mode, util::core_file &file, int tracknum, const cdrom_file::track_info &info, const std::string &filename, uint32_t frameoffs, uint64_t outputoffs)
 {
 	if (mode == MODE_GDI)
 	{
-		int mode = 0, size = 2048;
-
-		switch (info.trktype)
-		{
-			case cdrom_file::CD_TRACK_MODE1:
-				mode = 4;
-				size = 2048;
-				break;
-
-			case cdrom_file::CD_TRACK_MODE1_RAW:
-				mode = 4;
-				size = 2352;
-				break;
-
-			case cdrom_file::CD_TRACK_MODE2:
-				mode = 4;
-				size = 2336;
-				break;
-
-			case cdrom_file::CD_TRACK_MODE2_FORM1:
-				mode = 4;
-				size = 2048;
-				break;
-
-			case cdrom_file::CD_TRACK_MODE2_FORM2:
-				mode = 4;
-				size = 2324;
-				break;
-
-			case cdrom_file::CD_TRACK_MODE2_FORM_MIX:
-				mode = 4;
-				size = 2336;
-				break;
-
-			case cdrom_file::CD_TRACK_MODE2_RAW:
-				mode = 4;
-				size = 2352;
-				break;
-
-			case cdrom_file::CD_TRACK_AUDIO:
-				mode = 0;
-				size = 2352;
-				break;
-		}
+		const int tracktype = info.trktype == cdrom_file::CD_TRACK_AUDIO ? 0 : 4;
 		const bool needquote = filename.find(' ') != std::string::npos;
 		const char *const quotestr = needquote ? "\"" : "";
-		file.printf("%d %d %d %d %s%s%s %d\n", tracknum+1, frameoffs, mode, size, quotestr, filename, quotestr, discoffs);
+		file.printf("%d %d %d %d %s%s%s %d\n", tracknum+1, frameoffs, tracktype, info.datasize, quotestr, filename, quotestr, outputoffs);
 	}
 	else if (mode == MODE_CUEBIN)
 	{
-		// first track specifies the file
-		if (tracknum == 0)
+		// specify a new file when writing to the beginning of a file
+		if (outputoffs == 0)
 			file.printf("FILE \"%s\" BINARY\n", filename);
```

Comment:
I know the code never attempted to deal with it, but how are double quote characters supposed to be escaped in filenames in cuesheets?  Does it use CSV-style `""` to represent one `"` character inside a quoted string, or C-style backslash escapes, or something else?

Reply:
Generally unsupported — https://hydrogenaud.io/index.php/topic,87257.msg747275.html#msg747275

Reply:
I can't see there being a lot of demand for quotation marks in the exported filenames but it definitely could cause problems depending on implementation it seems. I wouldn't be against disallowing quotation marks in output bin filenames.

Reply:
Added an error for quotation marks in output bin filename.
https://github.com/mamedev/mame/pull/12191/commits/9b6d4bd5b22976c7788f86d0b2807ed0904bbfa5

```
> ./chdman extractcd -i test.chd  -o test_split.cue -f -sb -ob 'test".bin'
chdman - MAME Compressed Hunks of Data (CHD) manager 0.264 (mame0252-4087-gbed1cd5180a-dirty)
Output bin filename (test") must not contain quotation marks
Fatal error occurred: 1
```